src/sdk/modules/validators/smartSessionValidator/smartSessionValidator.test.ts

```diff
@@ -0,0 +1,324 @@
+      sessionKeyData: toHex(toBytes(sessionKeyEOA)),
+      sessionValidAfter: 0,
+      sessionValidUntil: 0,
+      actionPoliciesInfo: [
+        {
+          contractAddress: TEST_CONTRACTS.Counter.address, // counter address
+          functionSelector: "0x273ea3e3" as Hex, // function selector for increment count
+          validUntil: 0, // 1717001666
+          validAfter: 0,
+          rules: [], // no other rules and conditions applied
+          valueLimit: BigInt(0)
+        }
+      ]
+    }
+
+    const smartSessionNexusClient = nexusClient.extend(
+      smartSessionValidatorActions()
+    )
+
+    const enableSessionsResponse = await smartSessionNexusClient.enableSessions(
+      {
+        account: nexusClient.account,
+        sessionRequestedInfo: [sessionRequestedInfo]
+      }
+    )
+
+    expect(enableSessionsResponse.userOpHash).toBeDefined()
+    expect(enableSessionsResponse.permissionIds).toBeDefined()
+
+    const permissionIds = enableSessionsResponse.permissionIds
+    expect(permissionIds.length).toBe(1)
+    const permissionId = permissionIds[0]
+    cachedPermissionId = permissionId
+
+    const receipt = await nexusClient.waitForUserOperationReceipt({
+      hash: enableSessionsResponse.userOpHash
+    })
+
+    expect(receipt.success).toBe(true)
+
+    const isEnabled = await isSessionEnabled({
+      client: nexusClient.account.client as PublicClient,
+      accountAddress: nexusClient.account.address,
+      permissionId: permissionId
+    })
+    expect(isEnabled).toBe(true)
+  }, 60000)
+
+  test("should make use of already enabled session (USE mode) to increment a counter using a session key", async () => {
+    const isEnabled = await isSessionEnabled({
+      client: nexusClient.account.client as PublicClient,
+      accountAddress: nexusClient.account.address,
+      permissionId: cachedPermissionId
+    })
+    expect(isEnabled).toBe(true)
+
+    const smartSessionValidator = await toSmartSessionValidatorModule({
+      client: nexusClient.account.client as PublicClient,
+      initData: encodePacked(["address"], [eoaAccount.address]),
+      deInitData: "0x",
+      nexusAccountAddress: nexusClient.account.address,
+      activePermissionId: cachedPermissionId
+    })
+
+    const pubClient = nexusClient.account.client as PublicClient
+
+    const counterBefore = await pubClient.readContract({
+      address: TEST_CONTRACTS.Counter.address,
+      abi: CounterAbi,
+      functionName: "getNumber",
+      args: []
+    })
+
+    // helpful for out of range test
+    // await testClient.setNextBlockTimestamp({
+    //   timestamp: 9727001666n
+    // })
+
+    // set active validation module
+    nexusClient.account.setActiveValidationModule(smartSessionValidator)
+
+    const smartSessionNexusClient = nexusClient.extend(
+      smartSessionValidatorActions()
+    )
+
+    const userOpHash = await smartSessionNexusClient.useEnabledSession({
+      account: nexusClient.account,
```

Comment:
Why pass in account here if you already have the nexusClient. If you have nexusClient then you can find the account under the hood

Reply:
I have followed pattern from other decorators. 
GetSmartAccountParameter
 has this required parameter account

Reply:
Ok, let's tackle the devEx for both modules in a different ticket then. gabi already has one I believe. Happy to merge this then but let's focus on devEx in a different ticket before we officially release anything. 

The devEx right now doesn't quite make sense in places, but I think it's just a small task to get it right.